crates/hyperdrive-test-utils/src/chain/deploy.rs

```diff
@@ -949,20 +949,36 @@ impl TestnetDeploy for Chain {
                 .await?
                 .await?
                 .unwrap();
-            let logs = tx
-                .logs
-                .into_iter()
-                .filter_map(|log| {
-                    if let Ok(HyperdriveFactoryEvents::DeployedFilter(log)) =
-                        HyperdriveFactoryEvents::decode_log(&log.into())
-                    {
-                        Some(log)
-                    } else {
-                        None
+
+            // Retry loop in case Anvil fails.
+            let mut num_retries = 0;
+            loop {
+                let logs = tx
+                    .logs
+                    .clone()
```

Comment:
You need the `clone` because `into_iter` takes ownership of `tx.logs` without the `clone`. Previously, we didn't need this because `into_iter` was only being called once. Now it's looped over, so you can't take ownership here.

Reply:
ahh it's becuase of the loop. Makes sense, thanks!